[CI] ILMHistoryItemTests testTruncateLongError failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7138 / windows-2019_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/godusbo57mk52)
- [elasticsearch-periodic-platform-support #7138 / windows-2022_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/5iqfimurivpas)
- [elasticsearch-periodic #7116 / openjdk17_checkpart2_java-fips-matrix](https://gradle-enterprise.elastic.co/s/fiak6o3w523su)
- [elasticsearch-periodic #7116 / openjdk17_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/s6wf3j4gt5rwu)
- [elasticsearch-periodic #7116 / graalvm-ce17_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/lnexve6hxugxi)
- [elasticsearch-periodic-platform-support #7124 / windows-2019_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/bx73i2enkz5c4)
- [elasticsearch-periodic-platform-support #7124 / windows-2022_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/nbtv6yhw5l7xy)
- [elasticsearch-periodic #7102 / openjdk17_checkpart2_java-fips-matrix](https://gradle-enterprise.elastic.co/s/pbivx5s7b753e)
- [elasticsearch-periodic #7102 / openjdk17_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/cdjmvupsdbsse)
- [elasticsearch-periodic #7102 / graalvm-ce17_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/zytqy4ndskh4s)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ilm:test" --tests "org.elasticsearch.xpack.ilm.history.ILMHistoryItemTests.testTruncateLongError" -Dtests.seed=FB314CB68F54F81E -Dtests.locale=asa-TZ -Dtests.timezone=America/Mazatlan -Druntime.java=17
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ilm.history.ILMHistoryItemTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testTruncateLongError),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "{\"type\":\"illegal_argument_exception\",\"reason\":\"iiKfktKSpRbcAoAQafqyhoSqnuBofOwtTaNuUTXPmRMYIJnHnYNkmkzzdcIJTcKjmdYyAVZmGRJOjRzrAgxPtsZKtcPNEvDyfFUisCLRXdFACTjLTNanxmqFxqJxCddrYamcQbjuNKwthDynxwxoHMpQuqjisKUtAlwBZJlonbWwthsYlodHZZxnBOdlwfMbcSjUVhmFUOZMyPiqhYyDloGICseKyiTMCPpcouRCansfkvsYhIVDFfDOsNHdUiZGiegoBxVWnCqDFSEXlLpcnOrYzyJaknwdsNWuuUlSSAZhVCqJSfukVlOvlmTDnRHoSftYlLLXscJuWKfJoXhGppZfoCLntuOOFobAzwSKOltbwvIbCUrAvYTxmSBXdWrPLuFAvJizITQozThiYZOQkhOUurCnCqoRXRETYxvzYhsgSbyXnIulVToSArDXKObubuDJgBKRzbRgpZFeLjXwaCEquluLfRPdovDydlAWqUUIjgGuhhCHGONxjfSFTBjhQCDHbGrgjkvdVKXoNueXVugUkVSsaWSsbYxvkuCsEhEiwxICpuBdYehWCBgCmXSnOSXOnKHzfoZRGMoWVHnGAeMQrICrgESnNtqPGzkMlxfnNlKXMEpUbstKSRDWpbRzYOZUvTQFBTYpaUstMTfiQGmTAWRpYMIcqLFVjaBEBNnTNacDZZcUohuKqrwRxbSkIeHxxKIJEadqLclkeLTQEjmYGOBdpuXpfpOGrfikLwFcbhrFlmRSIiywjlNyKVvUEdjZoeyuwKAjTzaDzBsKRXoETKFlAhMaHrFGRrDQRlWllMdefcRMRXgQXvCURMFjIGuSRwgLwzgOuawhvjHtyTMPajoMrGOcHxexSCQYxCBZzdBLiUcHVsSHUtsuJxDQUFRLpXWDMYvS
[truncated]
```

**Issue Reasons:**
- [8.x] 5 consecutive failures in test testTruncateLongError
- [8.x] 3 consecutive failures in step windows-2019_checkpart2_platform-support-windows
- [8.x] 3 consecutive failures in step windows-2022_checkpart2_platform-support-windows
- [8.x] 3 consecutive failures in step openjdk17_checkpart2_java-fips-matrix
- [8.x] 3 consecutive failures in step openjdk17_checkpart2_java-matrix
- [8.x] 3 consecutive failures in step graalvm-ce17_checkpart2_java-matrix
- [8.x] 15 failures in test testTruncateLongError (14.7% fail rate in 102 executions)
- [8.x] 3 failures in step windows-2019_checkpart2_platform-support-windows (100.0% fail rate in 3 executions)
- [8.x] 3 failures in step windows-2022_checkpart2_platform-support-windows (100.0% fail rate in 3 executions)
- [8.x] 3 failures in step openjdk17_checkpart2_java-fips-matrix (100.0% fail rate in 3 executions)
- [8.x] 3 failures in step openjdk17_checkpart2_java-matrix (100.0% fail rate in 3 executions)
- [8.x] 3 failures in step graalvm-ce17_checkpart2_java-matrix (100.0% fail rate in 3 executions)
- [8.x] 3 failures in pipeline elasticsearch-periodic-platform-support (75.0% fail rate in 4 executions)
- [8.x] 3 failures in pipeline elasticsearch-periodic (75.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
java.lang.AssertionError: 	
Expected: "{\"type\":\"illegal_argument_exception\",\"reason\":\"yQSAQIzcPDGQUJZVsRcDPAfRuhXsabemZfhdOBUgCRPPMjAAGhxYnUhOrKxDgJcQlYdVgdUCTwljjrJyWxVONmaFijXlblhpnBIMcSGBFUBsbrmVgrREIvoqCsEjznNQOAXDMirpajDGGbIbUaJcrausOnDjZOGlLwMpwWklsINQjJvGsjvHpkoPfxLtjdKfrtsyyjHKVkJTqMyshNubvgEqwljampMmyYKsBtEDCFlQXVithgSzGiGhkgIRjAiJmmdJBeeTiVWMQKKkxwojwBWafzoTVAbIVzbZIKXVhhLyoPjiPdhpwiluYZgdVnfRCopXOSBLQPTgnoLZyIAZuhZLWkBIIGyKuaGTfrOJYsetiRrFZhnrzMhSurumJfkwwTyjFdOzfeJVREMhEahMkTnzetTWBZIlkBdENFFgooxmUgUcswbdQoVtDtVeofDDaRSpogygrerqAAMQEEAuqZKeeLAODPsPuhrhPUGfoCCELdXdJIpSguZWqhTftXlavDvuzbaTCXDlerfOTjzFBHKpidvaDsHaDEdSEnWDPjrqXPQrMrVVqLfSFbghALnxYyfimGBviXLdjesQItCFVDUArNYNpxPwlsBqzBZCiAsouIOneLHislvYzWfmMjHPWGjbKQfwiJpKsSVpkcvwbZsWAPVbnWzUtjOxgvhZypWgRJqXDpligxyCOzZfhsSWUYVMEqwTJmXgjDnlWmBZlwwARDsuuAYEscNurOLhEOpBVPUrVmNRCpDxEaQqkRoSwxOzZvuqneGZaDyMVyCjgCnkdOaKpbSigtLhaxGGQiqozFKILRxydRxwnuwgnFbyZXwEnfCQZeyhWfYJSYfWfIQwRBbLRUqydkSMHiNJbZSLoeWItTopZoMdoLeTtFyeyIcbMfbFuxhIyVRCAfSHlqpegfEbDwbIMAeAywNZfWdXXtAPCIJgDjARPNYWqDzTFiBvoeHteSOCrKKwB... (5126 chars truncated)\"}"	
     but: was "{\"type\":\"illegal_argument_exception\",\"reason\":\"yQSAQIzcPDGQUJZVsRcDPAfRuhXsabemZfhdOBUgCRPPMjAAGhxYnUhOrKxDgJcQlYdVgdUCTwljjrJyWxVONmaFijXlblhpnBIMcSGBFUBsbrmVgrREIvoqCsEjznNQOAXDMirpajDGGbIbUaJcrausOnDjZOGlLwMpwWklsINQjJvGsjvHpkoPfxLtjdKfrtsyyjHKVkJTqMyshNubvgEqwljampMmyYKsBtEDCFlQXVithgSzGiGhkgIRjAiJmmdJBeeTiVWMQKKkxwojwBWafzoTVAbIVzbZIKXVhhLyoPjiPdhpwiluYZgdVnfRCopXOSBLQPTgnoLZyIAZuhZLWkBIIGyKuaGTfrOJYsetiRrFZhnrzMhSurumJfkwwTyjFdOzfeJVREMhEahMkTnzetTWBZIlkBdENFFgooxmUgUcswbdQoVtDtVeofDDaRSpogygrerqAAMQEEAuqZKeeLAODPsPuhrhPUGfoCCELdXdJIpSguZWqhTftXlavDvuzbaTCXDlerfOTjzFBHKpidvaDsHaDEdSEnWDPjrqXPQrMrVVqLfSFbghALnxYyfimGBviXLdjesQItCFVDUArNYNpxPwlsBqzBZCiAsouIOneLHislvYzWfmMjHPWGjbKQfwiJpKsSVpkcvwbZsWAPVbnWzUtjOxgvhZypWgRJqXDpligxyCOzZfhsSWUYVMEqwTJmXgjDnlWmBZlwwARDsuuAYEscNurOLhEOpBVPUrVmNRCpDxEaQqkRoSwxOzZvuqneGZaDyMVyCjgCnkdOaKpbSigtLhaxGGQiqozFKILRxydRxwnuwgnFbyZXwEnfCQZeyhWfYJSYfWfIQwRBbLRUqydkSMHiNJbZSLoeWItTopZoMdoLeTtFyeyIcbMfbFuxhIyVRCAfSHlqpegfEbDwbIMAeAywNZfWdXXtAPCIJgDjARPNYWqDzTFiBvoeHteSOCrKKwB... (5208 chars truncated)\"}"
```
Only the number of chars seems to mismatch.

All four [failures](https://es-delivery-stats.elastic.dev/app/r/s/EoWO1)  occurred on Windows, that might be a clue? It doesn't reproduce for me locally.

Assigning to @dakrone as it's caused by #125054

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -338,9 +338,6 @@ tests:
 - class: org.elasticsearch.index.engine.ThreadPoolMergeSchedulerTests
   method: testSchedulerCloseWaitsForRunningMerge
   issue: https://github.com/elastic/elasticsearch/issues/125236
-- class: org.elasticsearch.xpack.ilm.history.ILMHistoryItemTests
-  method: testTruncateLongError
-  issue: https://github.com/elastic/elasticsearch/issues/125216
 - class: org.elasticsearch.index.shard.StoreRecoveryTests
   method: testAddIndices
   issue: https://github.com/elastic/elasticsearch/issues/124104

--- a/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/history/ILMHistoryItemTests.java
+++ b/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/history/ILMHistoryItemTests.java
@@ -22,7 +22,9 @@ import java.io.IOException;
 import java.util.Map;
 
 import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.matchesPattern;
 import static org.hamcrest.Matchers.startsWith;
 
 public class ILMHistoryItemTests extends ESTestCase {
@@ -135,15 +137,15 @@ public class ILMHistoryItemTests extends ESTestCase {
             try (XContentParser p = XContentFactory.xContent(XContentType.JSON).createParser(XContentParserConfiguration.EMPTY, json)) {
                 Map<String, Object> item = p.map();
                 assertThat(
-                    item.get("error_details"),
-                    equalTo(
+                    (String) item.get("error_details"),
+                    containsString(
                         "{\"type\":\"illegal_argument_exception\",\"reason\":\""
                             // We subtract a number of characters here due to the truncation being based
                             // on the length of the whole string, not just the "reason" part.
                             + longError.substring(0, LifecycleExecutionState.MAXIMUM_STEP_INFO_STRING_LENGTH - 47)
-                            + "... (5126 chars truncated)\"}"
                     )
                 );
+                assertThat((String) item.get("error_details"), matchesPattern(".*\\.\\.\\. \\(\\d+ chars truncated\\).*"));
             }
         }
     }

